hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/filtering/FilteredMessageHandler.java

```diff
@@ -41,6 +47,8 @@ public void handle(FilterResult result, Message message, Subscription subscripti
             if (subscription.isTrackingEnabled()) {
                 trackers.get(subscription).logFiltered(toMessageMetadata(message, subscription), result.getFilterType().get());
             }
+
+            consumerRateLimiter.acquire();
```

Comment:
Thats what i wrote in my comment - this is a risk we have to take now. I see no easy way to change it. The only way would be to have separate rate limiter that does not change - it's always set to subscription maxRate?